components/public-api/gitpod/experimental/v1/ide_workspace.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package gitpod.experimental.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1";
+
+service IDEWorkspaceService {
+    // SendHeartbeat sends a heartbeat signal to a running workspace.
+    rpc SendHeartbeat(SendHeartbeatRequest) returns (SendHeartbeatResponse) {}
+
+    // SendClose sends a close signal to a running workspace.
+    rpc SendClose(SendCloseRequest) returns (SendCloseResponse) {}
+}
+
+message SendHeartbeatRequest {
+    string instance_id = 1;
```

Comment:
As far as I understand the public API does not know/talk about workspace instances. Ie 'workspace_id' would be the appropriate choice. 

Cc @easyCZ

Reply:
For the `experimental` package of the Public API, this is less of a concern and I'm happy for us to also expose workspace instance.

Yes, ideally we'd not really talk about Workspace Instances in the Public API but in a way, that's not fully possible with existing WS APIs on Server. For now, what we do is we do expose the Workspace Instance ID but only in the Status response to a Workspace. From that perspective, it's in a way a secondary domain object, rather than being first class and customers having to understand them.

For the IDEWorkspaceService, which team IDE mostly intends to consume themselves at least for now I'm happy for us to continue iterating to find the right flow and get validation. If we did move this API to the `stable` package, we'd need to go through a review and think hard on how we name and structure the API, including exactly which fields do expose.

Reply:
One way to avoid the `instance_id` is to send a `workspace_id` and the WorkspaceIDEService performs a `WorkspaceService.GetWorkspace(workspace_id)` call to retrieve the `instance_id`.

Reply:
ok, thanks, it is good to realize it, maybe it is worth to document such key assumptions about API for future design

I guess we can go with @easyCZ suggestion now? We could also improve on it and do a check whether workspace running or not to prevent false positive heartbeats. Right now we have such logic both in VS Code browser and desktop to prevent. WDYT?

Reply:
I'm happy with it as is. We'd review it before going into "stable" anyway. Moving forward on this and getting validation is as important as trying to get it right (but ultimately we won't until we actually use it)